pkg/internal/ibmblockcsi/ibmblockcsi.go

```diff
@@ -67,7 +68,7 @@ func (c *IBMBlockCSI) GetLabels() labels.Set {
 }
 
 // GetAnnotations returns all the annotations to be set on all resources
-func (c *IBMBlockCSI) GetAnnotations() labels.Set {
+func (c *IBMBlockCSI) GetAnnotations(ds_restarted_key string , ds_restarted_value string) labels.Set {
```

Comment:
Functions should do one and only one thing.
In this case, the function should return the annotations.
Adding the key and value makes the function do two things.
Also, usually `get` functions don't change the data structure (`set functions do)

Reply:
I know it returns one thing.
It does more than one thing though.
For more information:
https://stackoverflow.com/questions/475675/when-is-a-function-too-long

Reply:
The function changes the DS. According to its name, it should be read-only (const function in C++)

Reply:
I don't think that it changes the DS. Its only returns the wanted annotations that should be on the DS/STS

Reply:
I meant DS == Data Structure, no Daemon Set.
Sorry for the confusion.

Reply:
done